metadata-ingestion/src/datahub/ingestion/source/identity/okta.py

```diff
@@ -316,7 +317,9 @@ def get_workunits(self) -> Iterable[MetadataWorkUnit]:
                 yield group_status_wu
 
         # Step 2: Populate GroupMembership Aspects for CorpUsers
-        datahub_corp_user_urn_to_group_membership: Dict[str, GroupMembershipClass] = {}
+        datahub_corp_user_urn_to_group_membership: Dict[
+            str, GroupMembershipClass
+        ] = defaultdict(lambda: GroupMembershipClass(groups=[]))
```

Comment:
The logic here is pretty complicated so I might be wrong, but I think this requires some changes below too. defaultdict makes dict[key] always succeed, but `key in dict` will still return false if the key hasn't been written to

Reply:
Should be fixed!